general/mem_manager.hpp

```diff
@@ -413,6 +414,38 @@ class Memory
    /** This method can be useful for debugging. It is explicitly instantiated
        for Memory<T> with T = int and T = double. */
    inline int CompareHostAndDevice(int size) const;
+
```

Comment:
When did you encountered the need to add this part?

Reply:
GCC 9.2 on Mac was giving warnings that we are using `new T[size]` with an over-aligned type when `T` is one of the SIMD types -- we were not using that case but the call to `new` was still there.